src/accel/LocalOffloadInterface.hh

```diff
@@ -0,0 +1,93 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/LocalOffloadInterface.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "corecel/Types.hh"
+#include "celeritas/Types.hh"
+
+class G4EventManager;
+
+namespace celeritas
+{
+struct SetupOptions;
+class SharedParams;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Abstract base class for offloading tracks to Celeritas.
+ *
+ * This class \em must be constructed locally on each worker
+ * thread/task/stream,
+ *
+ * \warning Due to Geant4 thread-local allocators, this class \em must be
+ * finalized or destroyed on the same CPU thread in which is created and used!
+ */
+class LocalOffloadInterface
+{
+  public:
+    // Initialize with options and core shared data
+    virtual void Initialize(SetupOptions const&, SharedParams&) = 0;
+
+    // Set the event ID and reseed the Celeritas RNG at the start of an event
+    virtual void InitializeEvent(int) = 0;
+
+    // Reseed the RNG states
+    virtual void Reseed(size_type) = 0;
```

Comment:
Is reseed intended to be called by the user or just part of the base interface? (If we refactor the InterfaceBase as suggested it can go away; if it needs to stay but is not intended to be aclled by the user, then just make it protected.)